File: FITCC.UI/mobile_fitcc/lib/Models/dogadjaj.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:mobile_fitcc/Models/agenda.dart';

part 'dogadjaj.g.dart';

@JsonSerializable(explicitToJson: true)
class Dogadjaj {
  int dogadjajId;
  String naziv;
  int trajanje;
  int agendaId;
  DateTime pocetak;
  DateTime kraj;
  String napomena;
  String lokacija;
  Agenda agenda;

  Dogadjaj(
     this.dogadjajId,
      this.naziv,
      this.trajanje,
      this.pocetak,
      this.kraj,
      this.napomena,
      this.agenda,
      this.agendaId,
      this.lokacija);

  factory Dogadjaj.fromJson(Map<String, dynamic> json) =>
      _$DogadjajFromJson(json);
  Map<String, dynamic> toJson() => _$DogadjajToJson(this);
}

File: FITCCRS2App.Models/Models/Dogadjaj.cs
﻿namespace FITCCRS2App.Models.Models
{
    public class Dogadjaj
	{
        public int DogadjajId { get; set; }
        public string? Naziv { get; set; }
        public int? Trajanje { get; set; }
        public DateTime? Pocetak { get; set; }
        public DateTime? Kraj { get; set; }
        public string? Napomena { get; set; }
        public string? Lokacija { get; set; }

        public int? AgendaId { get; set; }
        public virtual Agenda? Agenda { get; set; }
    }
}